README.md

```diff
@@ -342,20 +349,43 @@ for hiredis-cluster as well.
 
 ### Connecting
 
-The function `redisClusterAsyncConnect` can be used to
-establish a set of non-blocking connections to a Redis cluster.
-It returns a pointer to the newly created `redisClusterAsyncContext` struct.
-The `err` field should be checked after creation
-to see if there were errors creating the asynchronous cluster context.
+There are two alternative ways to initiate a cluster client which also determines
+how the client behaves during the initial connect.
+
+The first alternative is to use the function `redisClusterAsyncConnect`, which initially
+connects to the cluster in a blocking fashion and waits for the slot map before returning.
+Any command sent by the user thereafter will create a new non-blocking connection,
+unless a non-blocking connection already exists to the destination.
+The function returns a pointer to a newly created `redisClusterAsyncContext` struct.
+The `err` field should be checked after creation to make sure the slot map could be
+fetched correctly.
 
 ```c
 redisClusterAsyncContext *acc = redisClusterAsyncConnect("127.0.0.1:6379", HIRCLUSTER_FLAG_NULL);
 if (acc->err) {
     printf("Error: %s\n", acc->errstr);
-    // handle error
+    // handle error..
 }
+// ..or attach an event engine
+```
+
+The second alternative is to use `redisClusterAsyncContextInit` and `redisClusterAsyncConnect2`
+which avoids the initial blocking connect. This connection alternative requires an attached
+event engine when called but the initial connect and fetch of slot map is done in a
+non-blocking fashion.
```

Comment:
```suggestion
non-blocking fashion.

This means that commands sent directly after `redisClusterAsyncConnect2()` returns may fail because the initial slotmap has not yet been retrieved and then it is not yet known to which Redis node each command should be sent.
You may use the eventCallback to be notified when the slotmap is updated and the client is ready to accept commands.
```

It could be useful to have an event that's only fired once, where you can send your commands. You probably don't want to send them every time the slotmap is updated.
Thus, maybe we should have yet another event 'ready' or 'started' for when the client becomes ready for the first time, i.e. when the initial slotmap has been fetched. WDYT?

Do we also want to add an example of how to use the event callback to start sending commands using the async API? Putting acc as privdata... Or is that something the smart user can figure out?

Reply:
Great missed this info. Added a link to the ct_async.c testcase as well.

oh, a new event for initial slotmaps..well, it might simplify stuff.
At least it should be easy to implement; we can check for an initial update by using cc->table==NULL as an indicator that no update has yet been done in `updateNodesAndSlotmap()`.